libfido2/0001-cmake-msys-fixes.patch

```diff
@@ -0,0 +1,37 @@
+--- a/CMakeLists.txt	2020-02-19 20:51:59.000000000 +0330
++++ b/CMakeLists.txt	2020-02-27 23:59:57.303808300 +0330
+@@ -115,7 +115,7 @@
+ 	set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -pedantic")
+ 	set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -pedantic-errors")
+ 	set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fstack-protector-all")
+-	set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=c99")
++	#set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=c99")
```

Comment:
Could we add some information to the commit message why C99 fails?

I guess it is because of this:

```
[...]
cd /usr/src/MSYS2-packages/libfido2/src/libfido2-1.3.1/build-x86_64/src && /usr/bin/cc.exe -DHAS_DEV_URANDOM -DHAVE_ARC4RANDOM_BUF -DHAVE_ENDIAN_H -DHAVE_ERR_H -DHAVE_EXPLICIT_BZERO -DHAVE_GETENTROPY -DHAVE_GETLINE -DHAVE_GETOPT -DHAVE_GETPAGESIZE -DHAVE_SIGNAL_H -DHAVE_STRLCAT -DHAVE_STRLCPY -DHAVE_SYSCONF -DHAVE_TIMINGSAFE_BCMP -DHAVE_UNISTD_H -DSIGNAL_EXAMPLE -DTLS=__thread -D_FIDO_INTERNAL -D_FIDO_MAJOR=1 -D_FIDO_MINOR=3 -D_FIDO_PATCH=1 -D__BSD_VISIBLE=1 -Dfido2_shared_EXPORTS -I/usr/src/MSYS2-packages/libfido2/src/libfido2-1.3.1/src  -march=x86-64 -mtune=generic -O2 -pipe -Wall -Wextra -Werror -Wshadow -Wwrite-strings -Wmissing-prototypes -Wbad-function-cast -pedantic -pedantic-errors -fstack-protector-all -std=c99 -Wno-unused-result -Wcast-qual -O3 -DNDEBUG   -o CMakeFiles/fido2_shared.dir/assert.c.o   -c /usr/src/MSYS2-packages/libfido2/src/libfido2-1.3.1/src/assert.c
/usr/src/MSYS2-packages/libfido2/src/libfido2-1.3.1/src/assert.c: In function ‘fido_assert_verify’:
/usr/src/MSYS2-packages/libfido2/src/libfido2-1.3.1/src/assert.c:611:2: error: implicit declaration of function ‘explicit_bzero’ [-Wimplicit-function-declaration]
  611 |  explicit_bzero(buf, sizeof(buf));
      |  ^~~~~~~~~~~~~~
/usr/src/MSYS2-packages/libfido2/src/libfido2-1.3.1/src/assert.c: In function ‘fido_assert_set_rp’:
/usr/src/MSYS2-packages/libfido2/src/libfido2-1.3.1/src/assert.c:648:23: error: implicit declaration of function ‘strdup’; did you mean ‘strcmp’? [-Wimplicit-function-declaration]
  648 |  if ((assert->rp_id = strdup(id)) == NULL)
      |                       ^~~~~~
      |                       strcmp
/usr/src/MSYS2-packages/libfido2/src/libfido2-1.3.1/src/assert.c:648:21: error: assignment to ‘char *’ from ‘int’ makes pointer from integer without a cast [-Wint-conversion]
  648 |  if ((assert->rp_id = strdup(id)) == NULL)
      |                     ^
make[2]: *** [src/CMakeFiles/fido2_shared.dir/build.make:79: src/CMakeFiles/fido2_shared.dir/assert.c.o] Error 1
make[2]: Leaving directory '/usr/src/MSYS2-packages/libfido2/src/libfido2-1.3.1/build-x86_64'
make[1]: *** [CMakeFiles/Makefile2:102: src/CMakeFiles/fido2_shared.dir/all] Error 2
make[1]: Leaving directory '/usr/src/MSYS2-packages/libfido2/src/libfido2-1.3.1/build-x86_64'
make: *** [Makefile:133: all] Error 2
```

The reason is that these BSD functions require the preprocessor constant `_BSD_SOURCE` to be defined.

I also found that my copy of CMake (as of the MSYS2 package `cmake`) warned rather loudly:

```
  CMake no longer defines WIN32 on MSYS!
```

So I replaced this hunk with the following hunk and everything build just fine:

```diff
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9481c46..53d6cb6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -27,6 +27,11 @@ add_definitions(-D_FIDO_MAJOR=${FIDO_MAJOR})
 add_definitions(-D_FIDO_MINOR=${FIDO_MINOR})
 add_definitions(-D_FIDO_PATCH=${FIDO_PATCH})
 
+if (MSYS)
+	set(WIN32 on)
+	add_definitions(-D_BSD_SOURCE)
+endif()
+
 if(WIN32)
 	add_definitions(-DWIN32_LEAN_AND_MEAN)
 endif()
```

Reply:
This is really tricky, @cw789 started a conversation on libfido2 repository and @riedel purposed a PR fix on Yubico/libfido2#137 please check it and share your opinion.

Reply:
Too late, it was merged ;-)

But I did not see anything wrong there, I guess `_DEFAULT_SOURCE` implies `_BSD_SOURCE`.